src/desdemona/functions/sample_functions.clj

```diff
@@ -1,21 +1,23 @@
 (ns desdemona.functions.sample-functions
-  (:require [clojure
-             [walk :refer [postwalk]]]))
+  (:require
+   [clojure.core.match :refer [match]]))
 
 ;;; Defines functions to be used by the peers. These are located
 ;;; with fully qualified namespaced keywords, such as
-;;; desdemona.functions.sample-functions/format-line
+;;; desdemona.functions.sample-functions/add-message-origin
 
-(defn transform-segment-shape
-  "Recursively restructures a segment {:new-key [paths...]}"
-  [paths segment]
-  (try (let [f (fn [[k v]]
-                 (if (vector? v)
-                   [k (get-in segment v)]
-                   [k v]))]
-         (postwalk (fn [x] (if (map? x) (into {} (map f x)) x)) paths))
-       (catch Exception e
-         segment)))
+(defn build-row [segment]
+  {:line (str (name (segment :origin)) ": " (segment "MESSAGE"))})
```

Comment:
I agree that sooner is better than later. This is the first time we're actually doing anything with that data structure, so it'd be dumb to do it wrong for no reason. ;-)